public/scripts/modal.js

```diff
@@ -0,0 +1,18 @@
+export default async function initModal(document, domUtils) {
+  const $modalTrigger = document.querySelector('.popup-trigger');
+  const $popupModal = document.querySelector('.popup_modal');
+  $modalTrigger.addEventListener('click', (e) => {
+    domUtils.toggleVisibleElement($popupModal);
+    e.stopPropagation();
+  });
+
+  document.addEventListener('keyup', (e) => {
+    if (e.key === 'Escape') {
+      domUtils.hideElement($popupModal);
+    }
+  });
+
+  document.body.addEventListener('click', (e) => {
+    domUtils.hideElement($popupModal);
```

Comment:
Great, but if I click inside of the modal, it is closed, which seems rare:

![issue-16-2-01](https://user-images.githubusercontent.com/23049315/139817689-e4bcf1f0-46e5-4b5e-b12a-adc1f411b20d.gif)

The suggested change prevents it.

```suggestion
  const $popupBody = document.querySelector('.popup-body');
  $modalTrigger.addEventListener('click', (e) => {
    domUtils.toggleVisibleElement($popupModal);
    e.stopPropagation();
  });

  document.addEventListener('keyup', (e) => {
    if (e.key === 'Escape') {
      domUtils.hideElement($popupModal);
    }
  });

  document.body.addEventListener('click', (e) => {
    if (!e.path.includes($popupBody)) {
      domUtils.hideElement($popupModal);
    }
```